crates/noirc_evaluator/src/ssa/acir_gen/internal_var_cache.rs

```diff
@@ -6,12 +6,18 @@ use crate::{
     },
     Evaluator,
 };
-use acvm::FieldElement;
+use acvm::{
+    acir::native_types::{Expression, Witness},
+    FieldElement,
+};
 use std::collections::HashMap;
 
+use super::expression_to_witness;
+
 #[derive(Default)]
 pub(crate) struct InternalVarCache {
     inner: HashMap<NodeId, InternalVar>,
+    constants: HashMap<FieldElement, InternalVar>,
```

Comment:
Can you comment on what this? For example, does FieldElement just map to an internal var which holds an expression where q_c = FieldElement, or is it a witness that represents the constant?

Reply:
It is both, it depends. The internal var is first a constant expression, and it gets a witness once it is required (in a lazy manner).

Reply:
Taken from our call:

```
Doc comment: maps constant field elements to InternalVars. The internalVar may hold a witness, this is because some opcodes require Witnesses explicitly, for example the logic opcode.
```

Reply:
Done, I added a comment